docs/onboarding/basics/2-smart-contracts.md

```diff
@@ -0,0 +1,45 @@
+# Smart contracts
+
+Before we had Ethereum, we had different chains for different use cases; called app chains. Each chain is specifically dedicated to a single application/company. This was also in the early stages of crypto (think 2012-2014), and to be frank, these app chains were crap.
+
+The real innovation here was Vitalik realizing that we needed a more general execution environment, which wouldn't just execute logic associated with the specific app chain, but could support many third-party developers and very different applications.
+
+Appchains have gained popularity in the past few years in the Cosmos and Polkadot ecosystems; although these chains usually include smart contract facilities as well.
+
+[![Avoid the Middle-Man (Smart Contracts) - Computerphile](https://img.youtube.com/vi/csS1mZFuNSY/maxresdefault.jpg)](https://youtu.be/csS1mZFuNSY)
+
+### Takeaways
+
+- Understand how smart contracts are just rules written by humans.
+- Know smart contracts can only operate on data available onchain and are just limited in their capabilities.
+
+## Implementations of VMs
+
+The EVM is the most well-known smart contract engine (or VM). A virtual machine is a program, which can execute other programs. A smart contract is not magic at all, just a regular program; but instead of running on your computer, it is executed on every single node of a blockchain. Each node has to execute the contract to verify the correctness of a proposed block. An alternative to the EVM is webassembly, which was originally developed for the browser as an alternative to Javascript. Your browser executes code when it loads a website, code that has been developed by a third-party developer and thus cannot be trusted. The same problem exists for smart contracts. The blockchain cannot trust the provided code and must ensure a few things.
+
+- The code must be sandboxed. It may not access the actual node.
```

Comment:
`Access to the actual node from the code is very limited` would be more accurate?